fili-core/src/main/java/com/yahoo/bard/webservice/web/filters/HealthCheckFilter.java

```diff
@@ -40,12 +41,28 @@ public void filter(ContainerRequestContext requestContext) throws IOException {
             // See if we have any unhealthy checks
             Map<String, Result> unhealthyChecks = getUnhealthy();
             if (!unhealthyChecks.keySet().isEmpty()) {
+                StringBuilder debugMsgBuilder = new StringBuilder();
+                if (requestContext.getSecurityContext().getUserPrincipal() != null) {
+                    String user = requestContext.getSecurityContext().getUserPrincipal().getName();
+                    debugMsgBuilder.append("User=").append(user).append("\n");
+                }
+                debugMsgBuilder.append(renderUri(requestContext.getUriInfo().getRequestUri())).append("\n");
+
+                debugMsgBuilder.append("Timestamp: ");
+                debugMsgBuilder.append(java.time.Clock.systemUTC().instant().toString());
+                debugMsgBuilder.append("\n");
+
                 unhealthyChecks.entrySet()
-                        .forEach(entry -> LOG.error("Healthcheck '{}' failed: {}", entry.getKey(), entry.getValue()));
+                        .forEach(entry -> {
+                            LOG.error("Healthcheck '{}' failed: {}", entry.getKey(), entry.getValue());
+                        });
+                LOG.error(debugMsgBuilder.toString());
+
                 RequestLog.stopTiming(this);
+                debugMsgBuilder.insert(0, "Service is unhealthy. At least 1 healthcheck is failing\n");
```

Comment:
There needs a test to make sure `debugMsgBuilder` prints what's expected to avoid runtime surprises. You can put everything up to this point in a method for that